src/InfluxDB.LineProtocol/Payload/LineProtocolPoint.cs

```diff
@@ -37,40 +36,79 @@ public class LineProtocolPoint
             UtcTimestamp = utcTimestamp;
         }
 
-        public void Format(TextWriter textWriter)
+        public void Format(LineProtocolWriter writer)
         {
-            if (textWriter == null) throw new ArgumentNullException(nameof(textWriter));
+            if (writer == null) throw new ArgumentNullException(nameof(writer));
 
-            textWriter.Write(LineProtocolSyntax.EscapeName(Measurement));
+            writer.Measurement(Measurement);
 
             if (Tags != null)
             {
                 foreach (var t in Tags.OrderBy(t => t.Key))
                 {
-                    if (t.Value == null || t.Value == "")
+                    if (string.IsNullOrEmpty(t.Value))
                         continue;
 
-                    textWriter.Write(',');
-                    textWriter.Write(LineProtocolSyntax.EscapeName(t.Key));
-                    textWriter.Write('=');
-                    textWriter.Write(LineProtocolSyntax.EscapeName(t.Value));
+                    writer.Tag(t.Key, t.Value);
                 }
             }
 
-            var fieldDelim = ' ';
             foreach (var f in Fields)
             {
-                textWriter.Write(fieldDelim);
-                fieldDelim = ',';
-                textWriter.Write(LineProtocolSyntax.EscapeName(f.Key));
-                textWriter.Write('=');
-                textWriter.Write(LineProtocolSyntax.FormatValue(f.Value));
+                switch (f.Value)
+                {
+                    case sbyte value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case byte value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case short value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case ushort value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case int value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case uint value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case long value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case ulong value:
+                        if (value > long.MaxValue)
```

Comment:
`LineProtocolPoint` accepts ulong but as far as I can see influx cannot store a value that large.